File: back/src/main/java/com/a2m/back/service/sys/Sys0101Service.java
package com.a2m.back.service.sys;

import java.util.List;

import com.a2m.back.model.resp.DataResponse;
import com.a2m.back.model.resp.UserResponse;
import com.a2m.back.model.sys.EmpDto;

/**
 * Author tiennd
 * Created date 2023-07-17
 */
public interface Sys0101Service {
    int insert(EmpDto empDto) throws Exception;
    DataResponse diActiveUser (String USER_UID);
    List<UserResponse> getListUserInfo(String status,String page);
    List<UserResponse> searchByUserId(String userId);
    UserResponse getUserInfoByUserUid(String userUid);
    DataResponse getCountUser();
}

File: a2m-front/src/app/_service/sys/sys0101.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Cookie } from "ng2-cookies";
import { Observable } from 'rxjs';
import { AuthConstant } from 'app/_constant/auth.constant';
import { DataResponse } from 'app/_model/resp/data-response';
import { HeadersUtil } from 'app/_util/headers-util';
import { environment } from 'environments/environment';
import { UserInfo } from 'app/_model/auth/user-info';

@Injectable({
  providedIn: 'root'
})
export class Sys0101Service {

  constructor(
    private http: HttpClient
  ) { }


  getListUser(status: string, page: number): Observable<DataResponse> {
    const headers: HttpHeaders = HeadersUtil.getHeadersAuth();
    const url = environment.backApiUrl + `/admin/sys0101/listUserInfo/${status}?page=` + page;
    return this.http.get<DataResponse>(url, { headers: headers })
  }
  //   const url = environment.backApiUrl + `/admin/sys0301/phieumuon/${status}`;

  getUserByUserUid(userUid: String): Observable<UserInfo> {
    const headers: HttpHeaders = HeadersUtil.getHeadersAuth();
    const url = environment.backApiUrl + `/admin/sys0101/userInfo/${userUid}`;
    return this.http.get<DataResponse>(url, { headers: headers })
  }

  getCountUser(): Observable<DataResponse> {
    const headers: HttpHeaders = HeadersUtil.getHeadersAuth();
    const url = environment.backApiUrl + `/admin/sys0101/count`;
    return this.http.get<DataResponse>(url, { headers: headers });
  }

  getListUserByUserid(userId: string): Observable<DataResponse> {
    const headers: HttpHeaders = HeadersUtil.getHeadersAuth();
    const url = environment.backApiUrl + `/admin/sys0101/searchByUserId/${userId}`;
    return this.http.get<DataResponse>(url, { headers: headers })
  }
}